modules/snowflake-loader/src/main/scala/com/snowplowanalytics/snowplow/loader/snowflake/loading/SnowflakeLoader.scala

```diff
@@ -0,0 +1,87 @@
+package com.snowplowanalytics.snowplow.loader.snowflake.loading
+
+import cats.MonadThrow
+import cats.implicits._
+
+import com.snowplowanalytics.snowplow.analytics.scalasdk.SnowplowEvent
+import com.snowplowanalytics.snowplow.rdbloader.common.Common
+import com.snowplowanalytics.snowplow.rdbloader.discovery.{DataDiscovery, ShreddedType}
+import com.snowplowanalytics.snowplow.rdbloader.dsl.Logging
+import com.snowplowanalytics.snowplow.rdbloader.state.Control
+import com.snowplowanalytics.snowplow.rdbloader.algebras.db.TargetLoader
+import com.snowplowanalytics.snowplow.rdbloader.loading.Stage
+import com.snowplowanalytics.snowplow.rdbloader.LoaderError
+import com.snowplowanalytics.snowplow.loader.snowflake.db.SfDao
+import com.snowplowanalytics.snowplow.loader.snowflake.db.ast.AtomicDef
+import com.snowplowanalytics.snowplow.loader.snowflake.db.Statement.CopyInto
+import com.snowplowanalytics.snowplow.loader.snowflake.config.SnowflakeTarget
+
+class SnowflakeLoader[C[_]: MonadThrow: Logging: SfDao: Control](target: SnowflakeTarget)
+  extends TargetLoader[C] {
+
+  import SnowflakeLoader._
+
+  /**
+    * Run loading actions for atomic and shredded data
+    *
+    * @param discovery batch discovered from message queue
+    * @return block of statements to execute them out of a main transaction
+    */
+  def run(discovery: DataDiscovery): C[Unit] =
+    shreddedTypeCheck(discovery.shreddedTypes) match {
+      case Right(_) =>
+        val copyStatement = getStatement(discovery, target)
+        for {
+          _ <- Logging[C].info(s"Loading ${discovery.base}")
+          _ <- loadFolder(copyStatement)
+          _ <- Logging[C].info(s"Folder [${discovery.base}] has been loaded (not committed yet)")
+        } yield ()
+      case Left(err) =>
+        MonadThrow[C].raiseError(LoaderError.StorageTargetError(err))
+    }
+
+  def loadFolder(statement: CopyInto): C[Unit] =
+    Control[C].setStage(Stage.Loading("events")) *>
```

Comment:
Not necessary, but it would be nice to traverse the shredded types and set  loading `Stage`  to be consistent with redshift loader.

Reply:
I thought we are setting the stage with shredded types in Redshift Loader since it is loading the data to separate tables. Since we are loading to only one table in Snowflake Loader, I didn't add setting the stage for shredded types.